kadanes.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file
+ * @brief This program portrays a c++ implementation of the kadanes algorithm, 
+ * which finds the maximum subarray sum (https://www.geeksforgeeks.org/largest-sum-contiguous-subarray/)
+ * @details
+ *  The purpose of Kadanes algorithm is to dynamically traverse through every sum of contiguous subarrays and find the largest sum 
+ *  of positive contiguous segments. First, the program goes through every positive continuous segment (possible subarrays). The variable 
+ *  maxTemp keeps track of all the different sums of possible subarrays. If the subarray sum number goes below 0, it is set back to 0 because the 
+ *  biggest sum is never 0. (https://www.geeksforgeeks.org/largest-sum-contiguous-subarray/)
+ *  
+ * @author [Joseph Sulkes](https://github.com/jsulkes20)
+ *  
+ */
+#include<climits>
+#include<iostream>
+#include<vector>
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂